direct/nn/iterdualnet/iterdualnet.py

```diff
@@ -0,0 +1,201 @@
+        image_no_parameter_sharing: bool = True,
+        kspace_no_parameter_sharing: bool = True,
+        compute_per_coil: bool = True,
+        **kwargs,
+    ):
+        """Inits :class:`IterDualNet`.
+
+        Parameters
+        ----------
+        forward_operator : Callable
+            Forward Operator.
+        backward_operator : Callable
+            Backward Operator.
+        num_iter : int
+            Number of iterations. Default: 10.
+        image_normunet : bool
+            If True will use NormUNet for the image model. Default: False.
+        kspace_normunet : bool
+            If True will use NormUNet for the kspace model. Default: False.
+        image_no_parameter_sharing : bool
+            If False, a single image model will be shared across all iterations. Default: True.
+        kspace_no_parameter_sharing : bool
+            If False, a single kspace model will be shared across all iterations. Default: True.
+        compute_per_coil : bool
+            If True :math:`f` will be transformed into a multi-coil kspace.
+        kwargs : dict
+            Kwargs for unet models.
+        """
+        super().__init__()
+
+        self.forward_operator = forward_operator
+        self.backward_operator = backward_operator
+        self.num_iter = num_iter
+
+        self.image_no_parameter_sharing = image_no_parameter_sharing
+        self.kspace_no_parameter_sharing = kspace_no_parameter_sharing
+        image_unet_architecture = NormUnetModel2d if image_normunet else UnetModel2d
+        kspace_unet_architecture = NormUnetModel2d if kspace_normunet else UnetModel2d
+
+        self.image_block_list = nn.ModuleList()
+        self.kspace_block_list = nn.ModuleList()
+
+        for _ in range(self.num_iter if self.image_no_parameter_sharing else 1):
+            self.image_block_list.append(
+                image_unet_architecture(
+                    in_channels=2,
+                    out_channels=2,
+                    num_filters=kwargs.get("image_unet_num_filters", 8),
+                    num_pool_layers=kwargs.get("image_unet_num_pool_layers", 4),
+                    dropout_probability=kwargs.get("image_unet_dropout", 0.0),
+                )
+            )
+        for _ in range(self.num_iter if self.kspace_no_parameter_sharing else 1):
+            self.kspace_block_list.append(
+                kspace_unet_architecture(
+                    in_channels=2,
+                    out_channels=2,
+                    num_filters=kwargs.get("kspace_unet_num_filters", 8),
+                    num_pool_layers=kwargs.get("kspace_unet_num_pool_layers", 4),
+                    dropout_probability=kwargs.get("kspace_unet_dropout", 0.0),
+                )
+            )
+        self.compute_per_coil = compute_per_coil
+
+        self.lr = nn.Parameter(torch.ones(num_iter))
+        self.reg_param_I = nn.Parameter(torch.ones(num_iter))
+        self.reg_param_F = nn.Parameter(torch.ones(num_iter))
```

Comment:
Are these of interest externally or would local be fine?

Reply:
I don't understand the question